### PR TITLE
*: fix `Command` and `Time` in `show processlist`

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
When execute `show processlist`, TiDB only has `Query` Command state. 
According to MySQL doc, `Command` has many values: https://dev.mysql.com/doc/refman/8.0/en/thread-commands.html
For example,  id 1 in query result below should has `Sleep` state.

    +------+------+-----------+------+---------+------+-------+------------------+------+
    | Id   | User | Host      | db   | Command | Time | State | Info             | Mem  |
    +------+------+-----------+------+---------+------+-------+------------------+------+
    |    1 | root | 127.0.0.1 |      | Query   |   90 | 2     |                  |    0 |
    |    2 | root | 127.0.0.1 |      | Query   |    0 | 2     | show processlist |    0 |
    +------+------+-----------+------+---------+------+-------+------------------+------+
    2 rows in set (0.00 sec)

In MySQL:

    +----+------+-----------+------+---------+------+-------+------------------+
    | Id | User | Host      | db   | Command | Time | State | Info             |
    +----+------+-----------+------+---------+------+-------+------------------+
    |  1 | root | localhost | NULL | Sleep   |   10 |       | NULL             |
    |  2 | root | localhost | NULL | Query   |    0 | init  | show processlist |
    +----+------+-----------+------+---------+------+-------+------------------+
    2 rows in set (0.00 sec)

### What is changed and how it works?

- Add `Command` values.
- Set `Command` and `Time` before query/prepare/execute.
- Set `Time` to 0 after action finish.
- Set `Command` to `Sleep` after action finish.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -191,6 +193,10 @@ func (s *session) SetConnectionID(connectionID uint64) {
 	s.sessionVars.ConnectionID = connectionID
 }
 
+func (s *session) SetCommandValue(command byte) {
+	s.sessionVars.CommandValue = command
```
Comment (tiancaiamao) [inline session/session.go]: Is this thread safe ?

File: session/session.go [hunk 2]
```diff
@@ -703,12 +709,12 @@ func (s *session) ParseSQL(ctx context.Context, sql, charset, collation string)
...
 
-func (s *session) SetProcessInfo(sql string) {
+func (s *session) SetProcessInfo(sql string, t time.Time) {
 	pi := util.ProcessInfo{
 		ID:      s.sessionVars.ConnectionID,
 		DB:      s.sessionVars.CurrentDB,
-		Command: "Query",
-		Time:    time.Now(),
+		Command: mysql.Command2Str[s.sessionVars.CommandValue],
+		Time:    t,
```
Comment (tiancaiamao) [inline session/session.go]: Is `t` equal to `time.Now()` ?

File: session/session.go [hunk 3]
```diff
@@ -703,12 +709,12 @@ func (s *session) ParseSQL(ctx context.Context, sql, charset, collation string)
...
 }
 
-func (s *session) SetProcessInfo(sql string) {
+func (s *session) SetProcessInfo(sql string, t time.Time) {
 	pi := util.ProcessInfo{
 		ID:      s.sessionVars.ConnectionID,
 		DB:      s.sessionVars.CurrentDB,
-		Command: "Query",
-		Time:    time.Now(),
+		Command: mysql.Command2Str[s.sessionVars.CommandValue],
```
Comment (tiancaiamao) [inline session/session.go]: If we add `CommandValue` as a function parameter, could the the additional `sessionVars` be avoid ?

File: server/conn.go [hunk 4]
```diff
@@ -609,31 +616,43 @@ func (cc *clientConn) dispatch(data []byte) error {
 		if len(data) > 0 && data[len(data)-1] == 0 {
 			data = data[:len(data)-1]
 		}
+		cc.ctx.SetProcessInfo(hack.String(data), t)
```
Comment (tiancaiamao) [inline server/conn.go]: Will this leak the user's information, such as password ?

File: session/session.go (hunk 2 above)
Comment (imtbkcat) [reply session/session.go]: Sometimes no, so I use this parameter.

File: executor/adapter.go [hunk 5]
```diff
@@ -218,16 +215,8 @@ func (a *ExecStmt) Exec(ctx context.Context) (ast.RecordSet, error) {
 	var pi processinfoSetter
 	if raw, ok := sctx.(processinfoSetter); ok {
 		pi = raw
-		sql := a.OriginText()
-		if simple, ok := a.Plan.(*plannercore.Simple); ok && simple.Statement != nil {
-			if ss, ok := simple.Statement.(ast.SensitiveStmtNode); ok {
-				// Use SecureText to avoid leak password information.
-				sql = ss.SecureText()
```
Comment (lamxTyler) [inline executor/adapter.go]: Is the password protected now?

File: session/session.go (hunk 2 above)
Comment (tiancaiamao) [reply session/session.go]: When it's prepared statement ?

File: server/conn.go [hunk 6]
```diff
@@ -609,31 +611,41 @@ func (cc *clientConn) dispatch(data []byte) error {
 		if len(data) > 0 && data[len(data)-1] == 0 {
 			data = data[:len(data)-1]
 		}
+		cc.ctx.SetProcessInfo(hack.String(data), t, cmd)
```
Comment (zz-jason) [inline server/conn.go]: I think we should consider `ss.SecureText()`, as it does here:https://github.com/pingcap/tidb/pull/7844/files#diff-350127760839dbfd52d23927f7ff2d95L225

File: executor/adapter.go (hunk 5 above)
Comment (tiancaiamao) [reply executor/adapter.go]: I think it's important not to leak user's information.
Could we remove `cc.ctx.SetProcessInfo("", t, cmd)` in `dispatch` and do it here?
```
pi.SetProcessInfo(sql, t, CmdQuery)
```
@imtbkcat

**Comment by imtbkcat** [comment]
Security problem fixed. PTAL @tiancaiamao @zz-jason @lamxTyler

**Comment by tiancaiamao** [comment]
LGTM

File: executor/adapter.go [hunk 7]
```diff
@@ -259,9 +260,6 @@ func (a *ExecStmt) handleNoDelayExecutor(ctx context.Context, sctx sessionctx.Co
 
 	var err error
 	defer func() {
-		if pi != nil {
```
Comment (lamxTyler) [inline executor/adapter.go]: Seems `pi` is not used in this function?

File: executor/adapter.go [hunk 8]
```diff
@@ -118,9 +119,6 @@ func (a *recordSet) NewChunk() *chunk.Chunk {
 func (a *recordSet) Close() error {
 	err := a.executor.Close()
 	a.stmt.logSlowQuery(a.txnStartTS, a.lastErr == nil)
-	if a.processinfo != nil {
```
Comment (lamxTyler) [inline executor/adapter.go]: Is the `processinfo` used?

File: server/conn.go [hunk 9]
```diff
@@ -609,6 +623,7 @@ func (cc *clientConn) dispatch(data []byte) error {
 		if len(data) > 0 && data[len(data)-1] == 0 {
 			data = data[:len(data)-1]
 		}
+		// cc.ctx.SetProcessInfo(hack.String(data), t, cmd)
```
Comment (lamxTyler) [inline server/conn.go]: What is the comment for?

File: executor/adapter.go (hunk 8 above)
Comment (lamxTyler) [reply executor/adapter.go]: Is the two same?

Comment (imtbkcat) [reply executor/adapter.go]: Just by mistake, `pi` is not used in this function, so i delete it.

File: executor/adapter.go (hunk 7 above)
Comment (imtbkcat) [reply executor/adapter.go]: `pi` in `handleNoDelayExecutor` and `NewChunk()` is working for set command to `ComSleep` after sql execute. I delete both of them and move to here: https://github.com/pingcap/tidb/pull/7844/files/912d55e6d9b185ee5929159641dd8ee32a475053#diff-dc417f5322100e2e83a8818c70a25878R593

**Comment by imtbkcat** [comment]
PTAL @lamxTyler

File: server/conn.go (hunk 9 above)
Comment (lamxTyler) [reply server/conn.go]: This comment is not addressed. @imtbkcat

**Comment by imtbkcat** [comment]
unused comment addressed. @lamxTyler

**Pull request merged**